src/main/java/jenkins/plugins/mailer/tasks/MimeMessageBuilder.java

```diff
@@ -231,17 +265,28 @@ private void addRecipients(MimeMessage msg) throws UnsupportedEncodingException,
         addRecipients(msg, cc, Message.RecipientType.CC);
         addRecipients(msg, bcc, Message.RecipientType.BCC);
     }
+
     private void addRecipients(MimeMessage msg, Set<InternetAddress> recipientList, Message.RecipientType recipientType) throws UnsupportedEncodingException, MessagingException {
         if (recipientList.isEmpty()) {
             return;
         }
+        final Collection<InternetAddress> recipients = recipientFilter != null ? recipientFilter.apply(recipientList) : recipientList;
```

Comment:
It is a small refactor to share the code needed for the PR with the existing one.